src/backend/opengl/surface_impl.cpp

```diff
@@ -23,26 +23,27 @@
 using namespace gl;
 using namespace std;
 
-void generateGridIndices(unsigned short rows, unsigned short cols, unsigned short *indices)
+void generateGridIndices(unsigned short rows, unsigned short cols,
+                         std::vector<unsigned int>& indices)
 {
-unsigned short idx = 0;
-for(unsigned short r = 0; r < rows-1; ++r){
-    for(unsigned short c = 0; c < cols*2; ++c){
-        unsigned short i = c + (r * (cols*2));
-
-        if(c == cols * 2 - 1) {
-            *indices++ = idx;
-        }else{
-            *indices++ = idx;
-            if(i%2 == 0){
-                idx += cols;
-            } else {
-                idx -= (r%2 == 0) ? (cols-1) : (cols+1);
-            }
+    for (int r = 0; r < (rows-1); ++r) {
+        if (r > 0) {
+            // repeat first vertex for degenerate triangle
+            indices.push_back(r*rows);
+        }
+
+        for (int c = 0; c < cols; ++c) {
+            // One part of the strip
+            indices.push_back(r*rows + c);
+            indices.push_back((r+1)*rows + c);
+        }
+
+        if (r < (rows-2)) {
+            // repeat last vertex for degenerate triangle
+            indices.push_back(((r + 1) * rows) + (cols - 1));
```

Comment:
`push_back` it's great. Or at least without `reserve` it isnt. Can we reserve/resize before push_back?
@umar456 thoughts?